mergify_engine/merge_train.py

```diff
@@ -0,0 +1,510 @@
+                raise TrainCarPullRequestCreationFailure(self) from e
+
+        try:
+            pulls_for_title = ", ".join(
+                [f"#{p}" for p in self.parents + [self.pull_request_number]]
+            )
+            title = f"merge-queue: testing {self.initial_current_base_sha[:7]}, {pulls_for_title} together"
+            body = ""
+            tmp_pull = self.train.client.post(
+                f"/repos/{self.train.owner_login}/{self.train.repo_name}/pulls",
+                json={
+                    "title": title,
+                    "body": body,
+                    "base": self.train.ref,
+                    "head": branch_name,
+                },
+            ).json()
+        except http.HTTPClientSideError as e:
+            self._report_failure(e)
+            raise TrainCarPullRequestCreationFailure(self) from e
+
+        self.tmp_pull_request_number = tmp_pull["number"]
+
+        sub = asyncio.run(
+            subscription.Subscription.get_subscription(self.train.owner_id)
+        )
+        self.update_summaries(
+            context.Context(self.train.client, tmp_pull, sub),
+            None,
+            check_api.Conclusion.PENDING,
+            False,
+        )
+
+    def delete_pull(self) -> None:
+        if not self.tmp_pull_request_number:
+            return
+
+        branch = f"{constants.MERGE_QUEUE_BRANCH_PREFIX}/{self.train.ref}/{self.pull_request_number}"
+        try:
+            self.train.client.delete(
+                f"/repos/{self.train.owner_login}/{self.train.repo_name}/git/refs/heads/{branch}"
+            )
+        except http.HTTPNotFound:
+            pass
+
+    def _report_failure(self, exception: http.HTTPClientSideError) -> None:
+        # Update the original Pull Request
+        pull: github_types.GitHubPullRequest = self.train.client.item(
+            f"/repos/{self.train.owner_login}/{self.train.repo_name}/pulls/{self.pull_request_number}"
+        )
+
+        sub = asyncio.run(
+            subscription.Subscription.get_subscription(self.train.owner_id)
+        )
+        original_ctxt = context.Context(self.train.client, pull, sub, [])
+
+        title = "This pull request cannot tested to be merged"
+        summary = f"This pull request can't be tested to be merged because: {exception.message}"
+        original_ctxt.log.info(
+            "fail to create the pull request train car",
+            status=check_api.Conclusion.ACTION_REQUIRED,
+            title=title,
+            summary=summary,
+        )
+        check_api.set_check_run(
+            original_ctxt,
+            constants.MERGE_QUEUE_SUMMARY_NAME,
+            check_api.Result(
+                check_api.Conclusion.ACTION_REQUIRED,
+                title=title,
+                summary=summary,
+            ),
+        )
+
+        asyncio.run(github_events.send_refresh(original_ctxt.pull))
+
+    def update_summaries(
+        self,
+        tmp_pull_ctxt: context.Context,
```

Comment:
Not a fan of the `tmp` here. Can we find a better name?

Reply:
I rename some attributes, let me  known what you think.